---
title: "Evaluating Plant-Microbe Associations in Response to Environmental Stressors to Enhance Salt Marsh Restoration (microbiomes)"
author: "Kai Davis, Mary-Margaret McKinney, Rachel K. Gittman, Ariane L. Peralta"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
  pdf_document: default
---
# Setup
```{r setup, include=FALSE}
#use to set global options for chunks e.g., echo and warning options will be applied to all chunks:
knitr::opts_chunk$set(echo = TRUE)
# Clear environment:
rm(list=ls())

# Set working directory:
setwd("~/GitHub/2023_PSM_SMM_microbiomes/analyses")

# Standard error (se) and confidence interval (ci):
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code dependencies:
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
require("vegan"); require("reshape"); require("ggplot2"): require("tidyr"); require("dplyr"); require("dbstats"); require("reshape2"); require("ggpubr"); require("glue"); require("tidyverse"); require("labdsv"); require("phyloseq")
```

# Load files
*READ IN DESIGN FILE*
```{r Import Files - design, echo=TRUE}
file_path <- ("../data/PSM_SMM_design.csv")
design <- read.csv(file_path, row.names=1)
dim(design) #117  11
str(design)

#Convert chr to factors 
design$Study <- as.factor(design$Study)
design$Microbe_TRT <- as.factor(design$Microbe_TRT)
design$Water_TRT <- as.factor(design$Water_TRT)
design$Date2 <- as.factor(design$Date2)
design$TRT_row <- as.factor(design$TRT_row)
design$Source <- as.factor(design$Source) # coded as categorical for bulk vs rhiz
str(design)
```

# PROCESSING BACTERIAL SEQUENCES
*READ IN BACTERIAL SEQUENCING FILES*
```{r Import Files - Bacteria, include=FALSE}
#READ IN OTU FILE

otu_file <-("../data/PSM_SMM.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.opti_mcc.shared")
otu <- read.otu(otu_file)

#READ IN TAXONOMY FILE

tax_file <- ("../data/PSM_SMM.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.opti_mcc.0.03.cons.taxonomy")
tax <- import_mothur(mothur_constaxonomy_file =  tax_file)
colnames(tax)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax_df <- as.data.frame(tax)

otu.tax <- read.tax(taxonomy = "../data/PSM_SMM.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.opti_mcc.0.03.cons.taxonomy",
                   format = "rdp", tax.levels = 6, col.tax = 3)
tax_df <- as.data.frame(otu.tax)
```

*REMOVING SINGLETONS* 
```{r remove singletons, include=FALSE}
dim(otu) #117 71805

# OTU table - remove otus w/ < 2 occurrences across all sites
otu.2 <- otu[, which(colSums(otu) > 2)]
dim(otu.2) #117 38970

# OTU table - remove otus w/ < 10 occurrences across all sites
otu.10 <- otu[, which(colSums(otu) > 10)]
dim(otu.10) #117 20446
``` 
*TRANSFORM DATA*
```{r transform data, include=FALSE}
# Correct Sample IDs and Subset File
#check number of seqs per sample
#otu_sample <- rowSums(otu.2)
otu_sample <- rowSums(otu.10)
otu_sample

# OTU table - removed low abundance samples
otu_keep <- otu.10[which(rowSums(otu.10) >= 10000), ] #removed temporarily PSM_65 b/c it represents low reads from autoclaved field soil; SMM_08 failed PCR
dim(otu_keep) #removed three samples 

# OTU table - odd sites in bacterial composition data and remove in design file and remove positive control and remove all baseline samples
odd.sites <- c("Mock_Pos","PSM_65", "SMM_08", "PSM_61", "PSM_62","PSM_63", "PSM_64", "PSM_66", "PSM_67", "PSM_68")

otu_final <- otu_keep[setdiff(rownames(otu_keep), odd.sites), ]
design_final <- design[setdiff(rownames(design), odd.sites), ]
all.equal(rownames(design_final), rownames(otu_final))

dim(design_final) #107  11
design_PSM <- subset(design_final, Study == "PSM")
dim(design_PSM) #54 11
design_SMM <- subset(design_final, Study == "SMM")
dim(design_SMM) #53 11

otu_PSM <- otu_final[c(1:54),]
dim(otu_PSM) #54 20446
all.equal(rownames(design_PSM), rownames(otu_PSM))

otu_SMM <- otu_final[c(55:107),]
dim(otu_SMM) #53 20446
all.equal(rownames(design_SMM), rownames(otu_SMM))

#now that PSM and SMM are separated, we will apply removing rare OTUs that might have been missed
otu_PSM.10 <- otu_PSM[, which(colSums(otu_PSM) > 10)]
dim(otu_PSM.10) # 54 8331

otu_SMM.10 <- otu_SMM[, which(colSums(otu_SMM) > 10)]
dim(otu_SMM.10) # 53 14812

#RELATIVE ABUNDANCE
otu_rel_PSM <- otu_PSM.10
for(i in 1:dim(otu_PSM.10)[1]){
  otu_rel_PSM[i,] <- otu_PSM.10[i,]/sum(otu_PSM.10[i,])
}

otu_rel_SMM <- otu_SMM.10
for(i in 1:dim(otu_SMM.10)[1]){
  otu_rel_SMM[i,] <- otu_SMM.10[i,]/sum(otu_SMM.10[i,])
}
```

# ANALYSES OF BACTERIAL DIVERSITY AND COMMUNITY COMPOSITION

*ADD DIVERSITY METRICS - Plant-Soil-Micro (PSM) Experiment*
```{r Alpha diversity bacteria - PSM, echo=TRUE}
min(rowSums(otu_PSM.10)) #10941

otus.r <- rrarefy(otu_PSM.10, 10941)

# Fisher's Alpha
fisher <- fisher.alpha(otus.r)

# Species Richness
richness <- rowSums((otu_PSM.10 >= 1))

# Shannon Diversity
shannon <- diversity(otus.r, "shannon")

# Simpson's Evenness
simp.even <- apply(otus.r, 1, simp_even)

#Pielouâ€™s evenness
J <- shannon/log(specnumber(otus.r[,-c(1:1)]))

diversity_PSM <- cbind(design_PSM,richness,shannon,simp.even,J)
#write.csv(diversity,"../data/2014to2020.diversity.bact.raw.csv")

#ANOVA for Shannon diversity
  anova_result <- aov(shannon ~ Water_TRT * Microbe_TRT, data = diversity_PSM)
  summary(anova_result)

#ANOVA for Pielou's evenness
    anova_result <- aov(J ~ Water_TRT * Microbe_TRT, data = diversity_PSM)
  summary(anova_result)

str(diversity_PSM) 
dim(diversity_PSM) #54 15
```

*Plot Shannon diversity - PSM*
```{r Plot - Shannon Diversity PSM - field, echo=TRUE}
str(diversity_PSM)

# Graphing Shannon diversity
micro.labs <- c("No Addition", "Autoclaved Inocula", "Added Inocula") #for facet labels
names(micro.labs) <- c("CONTROL", "Microbe_neg", "Microbe_pos")

p <- ggplot(diversity_PSM, aes(x=Water_TRT,y=shannon, color=Water_TRT)) + 
  geom_boxplot() +
  geom_point(aes(color=Water_TRT), size=2, position = position_jitterdodge())+ 
  scale_color_manual(name="Treatment", 
                     values=c("#004AAD", "#FFBD59")) 

p1=p+facet_wrap(~Microbe_TRT)+facet_grid(. ~Microbe_TRT, labeller = labeller(Microbe_TRT=micro.labs))

shannon_PSM<-p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=1, hjust=1, 
          size=14), panel.border = element_rect(colour = "black",linewidth=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Treatment", y = "Shannon Diversity Index (H')") + 
    theme(strip.text.x = element_text(size=12, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", linewidth=1)) + theme(axis.text.x = element_text(angle = , hjust = ))

shannon_PSM
ggsave("../figures/PSM.shannon.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*Plot Pielou's Evenness - PSM*
```{r Plot - evenness - PSM, echo=TRUE}
# Graphing evenness

p <- ggplot(diversity_PSM, aes(x=Water_TRT,y=J, color=Water_TRT)) + 
  geom_boxplot() +
  geom_point(aes(color=Water_TRT), size=2, position = position_jitterdodge())+ 
  scale_color_manual(name="Treatment", 
                     values=c("#004AAD", "#FFBD59")) 

p1=p+facet_wrap(~Microbe_TRT)+facet_grid(. ~Microbe_TRT, labeller = labeller(Microbe_TRT=micro.labs))

evenness <- p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Treatment", y = "Pielou's Evenness (J')") +
    theme(strip.text.x = element_text(size=12, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", linewidth=1)) + theme(axis.text.x = element_text(angle = , hjust = )) +
    labs(color = "Treatment")
evenness
ggsave("../figures/PSM.evenness.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```
*PERMANOVA - PSM*
Permutational multivariate analysis of variance (PERMANOVA):
```{r PERMANOVA-PSM, echo=TRUE}
otu_design_PSM <- cbind(diversity_PSM,otu_rel_PSM)
dim(otu_design_PSM) #54 8346
dim(diversity_PSM) #54 15
otu_design_PSM <- droplevels(otu_design_PSM)
str(otu_design_PSM)

adonis_PSM <- adonis2(formula=otu_design_PSM[,-c(1:15)]~Water_TRT*Microbe_TRT, 
                     method = "bray", 
                     data = otu_design_PSM, 
                     perm=1000,
                     by = "terms",
                     set.seed=42)
adonis_PSM
```

*Graphing PCoA - PSM*
```{r PCoA: PSM, echo=TRUE}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
PSM_BC <- vegdist(otu_design_PSM[-c(1:15)], method="bray") #deleted last three samples which are baseline soils

pcoa_PSM <- cmdscale(PSM_BC, k=2, eig=TRUE, add=TRUE) # changing k to 2 because we only want 2 dimensions anyway
  # Now trying add = TRUE because of https://www.youtube.com/watch?v=G5Qckqq5Erw

explainvar1b_PSM <- round(pcoa_PSM$eig[1] / sum(pcoa_PSM$eig), 3) * 100
explainvar2b_PSM <- round(pcoa_PSM$eig[2] / sum(pcoa_PSM$eig), 3) * 100
sum.eigb <- sum(explainvar1b_PSM, explainvar2b_PSM)

explainvar1b_PSM # 21.5
explainvar2b_PSM # 11.7

otu_design_PSM <- otu_design_PSM[c(1:54),]
pcoa.groups <- paste(otu_design_PSM$Water_TRT, otu_design_PSM$Microbe_TRT,sep = "_")
pcoa.points <- data.frame(pcoa_PSM$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

Microbe <- c("No Addition", "Autoclaved Inocula", "Added Inocula","No Addition", "Autoclaved Inocula", "Added Inocula")
Water <- c("FRESH","FRESH","FRESH","SALT","SALT","SALT")

pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$Microbe <- as.factor(Microbe)
pcoa.cent.dataframe.trts$Water <- as.factor(Water)
view(pcoa.cent.dataframe.trts)
df1a <- pcoa.cent.dataframe.trts

#Plot
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=Water, shape = Microbe)) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
  theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + 
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") +
  geom_point(aes(colour=Water), size=5, stroke = 1.25, show.legend = TRUE) +
 scale_color_manual(name="Treatment", 
                     values=c("#004AAD", "#FFBD59")) +
  theme(axis.title = element_text(size=14), 
        axis.text = element_text(size=14),
        axis.text.x = element_text(size=14), 
        panel.border = element_rect(colour = "black", linewidth = 1.25)) + 
  theme(axis.ticks.length = unit(0.3, "cm")) + 
 xlab("PCoA 1 (21.5%)") + ylab("PCoA 2 (11.7%)") +
        labs(colour = "Water", shape = "Microbe") +
    guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
           shape = guide_legend(override.aes = list(size = 4))) +
    ggtitle("Bacterial 16SrRNA Community Composition")

ggsave("../figures/PSM.bact.ordination.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

```{r PCoA - PSM indiv, echo=TRUE}
#PSM_BC <- vegdist(otu_design_PSM[c(1:54),-c(1:15)], method="bray") #deleted last three samples which are baseline soils

##pcoa_PSM <- cmdscale(PSM_BC, k=2, eig=TRUE, add=TRUE) # changing k to 2 because we only want 2 dimensions anyway
  # Now trying add = TRUE because of https://www.youtube.com/watch?v=G5Qckqq5Erw

#explainvar1b_PSM <- round(pcoa_PSM$eig[1] / sum(pcoa_PSM$eig), 3) * 100
#explainvar2b_PSM <- round(pcoa_PSM$eig[2] / sum(pcoa_PSM$eig), 3) * 100
#sum.eigb <- sum(explainvar1b_PSM, explainvar2b_PSM)

#explainvar1b_PSM # 21.5
#explainvar2b_PSM # 11.7

pcoa_PSM.groups <- paste(otu_design_PSM$Water_TRT, otu_design_PSM$Microbe_TRT, sep = "_")
pcoa_PSM.points <- data.frame(pcoa_PSM$points, group = pcoa_PSM.groups)

pcoa.PSM.cent.dataframe.trts <- as.data.frame(pcoa_PSM.points)
#"No Addition", "Autoclaved Inocula", "Added Inocula"
Microbe <- c("Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula","Added Inocula", "Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","Autoclaved Inocula","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition","No Addition")
Water <- c("FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","SALT","SALT","SALT","SALT","SALT","SALT","SALT","SALT","SALT","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","SALT","SALT","SALT","SALT","SALT","SALT","SALT","SALT","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","FRESH","SALT","SALT","SALT","SALT","SALT","SALT","SALT","SALT","SALT")

pcoa_PSM_indiv <- cbind(pcoa.PSM.cent.dataframe.trts,Water, Microbe)
str(pcoa_PSM_indiv)
pcoa_PSM_indiv$Water <- as.factor(pcoa_PSM_indiv$Water)
pcoa_PSM_indiv$Microbe <- as.factor(pcoa_PSM_indiv$Microbe)
dim(pcoa_PSM_indiv)
df1a <- pcoa_PSM_indiv
str(df1a)

#Plot using ggplot2 - individual

bact_PSM_indiv<- ggplot(df1a, aes(x=X1, y=X2, shape=Microbe, colour=Water)) +   
  geom_point(aes(shape = Microbe, colour = Water), stroke = 4) +
#removes gridlines from plot
theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black")) +
theme(axis.title = element_text(size=12),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", linewidth=1.25)) +
 scale_color_manual(name="Treatment", 
                     values=c("#004AAD", "#FFBD59")) +
  #Set plot title text size
theme(plot.title=element_text(size=12)) +
#Set legend text size
theme(legend.text=element_text(size=10, face="bold"), legend.title = element_text(size=10, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
 xlab("PCoA 1 (21.5%)") + ylab("PCoA 2 (11.7%)") +
labs(shape = "Microbe") +
labs(colour="Water") + 
guides(color = guide_legend(override.aes = list(size = 1))) 
#ggtitle("Hammocks Beach Bacterial 16SrRNA Community Composition")

bact_PSM_indiv
ggsave("../figures/PSM.bact.indiv.ordination.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*Bacterial community indicator species analysis - PSM*
```{r PSM Bacteria Indicator Species i, eval=FALSE, include=FALSE}
install.packages("indicspecies")
require("indicspecies")
#otu_design_PSM <- otu_design_PSM[c(1:54),]

# Water_TRT
levels(otu_design_PSM$Water_TRT)
group <- otu_design_PSM$Water_TRT
design.type <- group
dataREL.t <- otu_rel_PSM[, colSums(otu_rel_PSM) > 0.05]
bac.ind <- indval(dataREL.t, design.type)
levels(design.type) #"FRESH" "SALT" 
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tax_df[which(as.character(tax_df$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_Water_TRT_PSM.txt",
            sep="\t", row.names = F, quote = F)

# Microbe_TRT
levels(otu_design_PSM$Microbe_TRT)
group <- otu_design_PSM$Microbe_TRT
design.type <- group
dataREL.t <- otu_rel_PSM[, colSums(otu_rel_PSM) > 0.05]
bac.ind <- indval(dataREL.t, design.type)
levels(design.type) #"CONTROL"     "Microbe_neg" "Microbe_pos"
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.1)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tax_df[which(as.character(tax_df$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_Microbe_TRT_PSM.txt",
            sep="\t", row.names = F, quote = F)
```

*ADD DIVERSITY METRICS - Soil Marsh Microbes (SMM) field sampling*
```{r Alpha diversity bacteria - SMM field sampling, echo=TRUE}
library(vegan)

min(rowSums(otu_SMM.10)) #20108

otus.r <- rrarefy(otu_SMM.10, 20108)

# Fisher's Alpha
fisher <- fisher.alpha(otus.r)

# Species Richness
richness <- rowSums((otu_SMM.10 >= 1))

# Shannon Diversity
shannon <- vegan::diversity(otus.r, index = "shannon")
#shannon <- diversity(otus.r, index = "shannon") - error here

# Simpson's Evenness
simp.even <- apply(otus.r, 1, simp_even)

#Pielouâ€™s evenness
J <- shannon/log(specnumber(otus.r[,-c(1:1)]))

diversity_SMM <- cbind(design_SMM,richness,shannon,simp.even,J)
#write.csv(diversity,"../data/2014to2020.diversity.bact.raw.csv")

str(diversity_SMM) 
dim(diversity_SMM) #53 15
```

*Plot Shannon diversity - SMM*
```{r Plot - Shannon Diversity SMM - field, echo=TRUE}
str(diversity_SMM)
levels(diversity_SMM$Date2)
#[1] "1_June12" "2_July13" "3_Oct5" 

# Graphing Shannon diversity
p <- ggplot(diversity_SMM, aes(x=TRT_row, y=shannon, color=TRT_row, shape=Source)) +
  geom_point(aes(color=TRT_row), size=3, position = position_jitterdodge()) 
p1 <- p +facet_wrap(~Date2)

shannon <- p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Row location (decreasing elevation)", y = "Shannon Diversity Index (H')") +
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", linewidth =1)) + theme(axis.text.x = element_text(angle = )) +
    labs(color = "Row location", shape = "Source")
  #ggtitle("bacterial diversity - bulk sediment") + 
            #theme(plot.title = element_text(face = "italic")) +
            #theme(legend.position = "none")
shannon
ggsave("../figures/SMM.shannon.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*Plot Pielou's Evenness - SMM*
```{r Plot - evenness - SMM, echo=TRUE}
# Graphing evenness
p <- ggplot(diversity_SMM, aes(x=TRT_row, y=J, color=TRT_row, shape=Source)) +
  geom_point(aes(color=TRT_row), size=3, position = position_jitterdodge()) 
p1 <- p +facet_wrap(~Date2)

evenness <- p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Row location (decreasing elevation)", y = "Pielou's Evenness (J')") +
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", linewidth=1)) + theme(axis.text.x = element_text(angle = )) +
    labs(color = "Row location", shape = "Source")
evenness
ggsave("../figures/SMM.evenness.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*PERMANOVA Permutational multivariate analysis of variance (PERMANOVA)*
```{r PERMANOVA-SMM, echo=TRUE}
otu_design_SMM <- cbind(design_SMM,otu_rel_SMM)
dim(otu_design_SMM) #53 14823
dim(design_SMM) #53 11

#combined date and using Source as random effect
adonis_SMM_row = adonis2(otu_design_SMM[,-c(1:11)]~TRT_row, strata = otu_design_SMM$Source, method = "bray", data = otu_design_SMM, perm=1000, set.seed=42)
adonis_SMM_row
```
*PCoA - average*
```{r PCoA: by source, echo=TRUE}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
SMM_BC <- vegdist(otu_design_SMM[,-c(1:11)], method="bray")

pcoa_SMM <- cmdscale(SMM_BC, k=2, eig=TRUE, add=TRUE) # changing k to 2 because we only want 2 dimensions anyway
  # Now trying add = TRUE because of https://www.youtube.com/watch?v=G5Qckqq5Erw

explainvar1b_SMM <- round(pcoa_SMM$eig[1] / sum(pcoa_SMM$eig), 3) * 100
explainvar2b_SMM <- round(pcoa_SMM$eig[2] / sum(pcoa_SMM$eig), 3) * 100
sum.eigb <- sum(explainvar1b_SMM, explainvar2b_SMM)

explainvar1b_SMM # 34.4
explainvar2b_SMM # 11.1

pcoa.groups <- paste(otu_design_SMM$TRT_row, otu_design_SMM$Source,sep = "_")
pcoa.points <- data.frame(pcoa_SMM$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
row <- c("1","1","2","2","3","3","4","4","5","5","6","6","7","7","8","8","9","9")
source <- c("bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz","bulk","rhiz")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$row <- as.factor(row)
pcoa.cent.dataframe.trts$source <- as.factor(source)
view(pcoa.cent.dataframe.trts)
df1a <- pcoa.cent.dataframe.trts

#Plot
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=row, shape = source)) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
  theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + 
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") +
  geom_point(aes(colour=row), size=5, stroke = 1.25, show.legend = TRUE) +
  scale_shape_manual(labels = c("bulk","rhizosphere"), values = c(16,17)) +
  theme(axis.title = element_text(size=14), 
        axis.text = element_text(size=14),
        axis.text.x = element_text(size=14), 
        panel.border = element_rect(colour = "black", size = 1.25)) + 
  theme(axis.ticks.length = unit(0.3, "cm")) + 
        xlab("PCoA 1 (34.4%)") + ylab("PCoA 2 (11.1%)") +
        labs(colour = "Row location", shape = "Source") +
    guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
           shape = guide_legend(override.aes = list(size = 4))) +
    ggtitle("Bacterial 16SrRNA Community Composition")

ggsave("../figures/SMM.bact.ordination.mean.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*PCoA - SMM individual communities*
```{r PCoA - SMM indiv, echo=TRUE}
#SMM_BC <- vegdist(otu_design_SMM[,-c(1:11)], method="bray")

#pcoa_SMM <- cmdscale(SMM_BC, k=2, eig=TRUE, add=TRUE) 

#explainvar1b_SMM <- round(pcoa_SMM$eig[1] / sum(pcoa_SMM$eig), 3) * 100
#explainvar2b_SMM <- round(pcoa_SMM$eig[2] / sum(pcoa_SMM$eig), 3) * 100
#sum.eigb <- sum(explainvar1b_SMM, explainvar2b_SMM)

#explainvar1b_SMM # 34.4
#explainvar2b_SMM # 11.1

pcoa_SMM.groups <- paste(otu_design_SMM$TRT_row, otu_design_SMM$Source, sep = "_")
pcoa_SMM.points <- data.frame(pcoa_SMM$points, group = pcoa_SMM.groups)

pcoa.SMM.cent.dataframe.trts <- as.data.frame(pcoa_SMM.points)

# Adding columns with the factors. H: hydrologic history
row <- c("1", "2","3","4","5","6","7","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9") 
source <- c("bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz")

pcoa_SMM_indiv <- cbind(pcoa.SMM.cent.dataframe.trts,row, source)
str(pcoa_SMM_indiv)
pcoa_SMM_indiv$row <- as.factor(pcoa_SMM_indiv$row)
pcoa_SMM_indiv$source <- as.factor(pcoa_SMM_indiv$source)
dim(pcoa_SMM_indiv)
df1a <- pcoa_SMM_indiv
str(df1a)

#Plot using ggplot2 - individual no Date2

bact_SMM_indiv<- ggplot(df1a, aes(x=X1, y=X2, shape=source, colour=row)) +   
  geom_point(aes(shape = source, colour = row), size = 4) +
#removes gridlines from plot
theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black")) +
theme(axis.title = element_text(size=12),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title text size
theme(plot.title=element_text(size=12)) +
#Set legend text size
theme(legend.text=element_text(size=10, face="bold"), legend.title = element_text(size=10, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
 xlab("PCoA 1 (34.4%)") + ylab("PCoA 2 (11.1%)") +
labs(shape = "Source") +
labs(colour="Row location") + 
guides(color = guide_legend(override.aes = list(size = 1))) +
ggtitle("Hammocks Beach Bacterial 16SrRNA Community Composition")

bact_SMM_indiv
ggsave("../figures/SMM.bact.indiv.ordination.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*PCoA - SMM indiv - subset rows 1,5,9*
```{r PCoA - SMM individual rows 1, eval=FALSE, include=FALSE}
pcoa_SMM.points_row159 <- pcoa_SMM_indiv[pcoa_SMM_indiv$row %in% c("1", "5", "9"), ]

source_Date2 <-c(
  "1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz", 
  "2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz",
  "3_Oct5_bulk", "3_Oct5_bulk", "3_Oct5_bulk", "3_Oct5_rhiz", "3_Oct5_rhiz", "3_Oct5_rhiz")

df1a <- cbind(pcoa_SMM.points_row159,source_Date2)
df1a$source_Date2 <- as.factor(df1a$source_Date2)
str(df1a)

#Plot using ggplot2 - individual no Date2

bact_SMM_indiv_159 <- ggplot(df1a, aes(x=X1, y=X2, shape=source_Date2, colour=row)) +   
  geom_point(aes(shape = source_Date2, colour = row), size = 4, stroke=1) +
  scale_shape_manual(labels = c("1_June12_bulk","1_June12_rhiz","2_July13_bulk", "2_July13_rhiz", "3_Oct5_bulk", "3_Oct5_rhiz"), values = c(15, 0, 16, 1, 17, 2)) +
#removes gridlines from plot
theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black")) +
theme(axis.title = element_text(size=12),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title text size
theme(plot.title=element_text(size=12)) +
#Set legend text size
theme(legend.text=element_text(size=10, face="bold"), legend.title = element_text(size=10, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
 xlab("PCoA 1 (34.4%)") + ylab("PCoA 2 (11.1%)") +
labs(shape = "Source") +
labs(colour="Row location") + 
scale_colour_manual(labels = c("landward", "midpoint","seaward"),
                    values = c("#D81B60","#1E88E5", "#8D82C6")) +
guides(color = guide_legend(override.aes = list(size = 4))) + 
ggtitle("Hammocks Beach Bacterial 16SrRNA Community Composition")

bact_SMM_indiv_159
ggsave("../figures/SMM.bact.indiv.159.ordination.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*PCoA - SMM indiv Date2*
```{r PCoA - SMM indiv Date2, echo=TRUE}
#SMM_BC <- vegdist(otu_design_SMM[,-c(1:11)], method="bray")
#df_SMM_BC <- as.data.frame(SMM_BC)
#pcoa_SMM <- cmdscale(SMM_BC, k=2, eig=TRUE, add=TRUE) 

#explainvar1b_SMM <- round(pcoa_SMM$eig[1] / sum(pcoa_SMM$eig), 3) * 100
#explainvar2b_SMM <- round(pcoa_SMM$eig[2] / sum(pcoa_SMM$eig), 3) * 100
#sum.eigb <- sum(explainvar1b_SMM, explainvar2b_SMM)

#explainvar1b_SMM # 34.4
#explainvar2b_SMM # 11.1

pcoa_SMM.groups <- paste(otu_design_SMM$TRT_row, otu_design_SMM$Source, otu_design_SMM$Date2, sep = "_")
pcoa_SMM.points <- data.frame(pcoa_SMM$points, group = pcoa_SMM.groups)

pcoa.SMM.cent.dataframe.trts <- as.data.frame(pcoa_SMM.points)

# Adding columns with the factors.
row <- c("1", "2","3","4","5","6","7","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9","1", "2","3","4","5","6","7","8","9") 

source <- c("bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","bulk", "bulk","bulk","bulk","bulk","bulk","bulk","bulk","bulk","rhiz", "rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz","rhiz")

source_Date2 <-c("1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_bulk","1_June12_rhiz", "1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","1_June12_rhiz","2_July13_bulk", "2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_bulk","2_July13_rhiz", "2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","2_July13_rhiz","3_Oct5_bulk", "3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_bulk","3_Oct5_rhiz", "3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz","3_Oct5_rhiz")
  
pcoa_SMM_indiv_Date2 <- cbind(pcoa.SMM.cent.dataframe.trts,row, source, source_Date2)
str(pcoa_SMM_indiv_Date2)
pcoa_SMM_indiv_Date2$row <- as.factor(pcoa_SMM_indiv_Date2$row)
pcoa_SMM_indiv_Date2$source <- as.factor(pcoa_SMM_indiv_Date2$source)
pcoa_SMM_indiv_Date2$source_Date2 <- as.factor(pcoa_SMM_indiv_Date2$source_Date2)
dim(pcoa_SMM_indiv_Date2)
df2a <- pcoa_SMM_indiv_Date2
str(df2a)

#Plot using ggplot2 - individual with Date2

bact_SMM_indiv_Date2<- ggplot(df2a, aes(x=X1, y=X2, shape=source_Date2, colour=row)) +   
  geom_point(aes(shape = source_Date2, colour = row), size = 4) +
  scale_shape_manual(labels = c("1_June12_bulk","1_June12_rhiz","2_July13_bulk", "2_July13_rhiz", "3_Oct5_bulk", "3_Oct5_rhiz"),
                     values = c(15, 0, 16, 1, 17, 2)) +
#removes gridlines from plot
theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black")) +
theme(axis.title = element_text(size=12),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title text size
theme(plot.title=element_text(size=12)) +
#Set legend text size
theme(legend.text=element_text(size=10, face="bold"), legend.title = element_text(size=10, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
 xlab("PCoA 1 (34.1%)") + ylab("PCoA 2 (11.0%)") +
labs(shape = "Source") +
labs(colour="Row location") + 
guides(color = guide_legend(override.aes = list(size = 2))) +
ggtitle("Hammocks Beach Bacterial 16SrRNA Community Composition")

bact_SMM_indiv_Date2
ggsave("../figures/SMM.bact.indiv.Date.ordination.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

*Bacterial community indicator species analysis - SMM field*
```{r Bacteria Indicator Species i, eval=FALSE, include=FALSE}
#trying indicp spp using different package and function https://cran.r-project.org/web/packages/indicspecies/vignettes/IndicatorSpeciesAnalysis.html
require("indicspecies")

# Source
levels(otu_design_SMM$Source)
str(otu_design_SMM)
#levels(otu_design_SMM$Date2)
group <- otu_design_SMM$Source
design.type <- group
dataREL.t <- otu_rel_SMM[, colSums(otu_rel_SMM) > 0.1]
bac.ind <- indval(dataREL.t, design.type)
levels(design.type) #[1] "bulk" "rhiz"
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.1)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tax_df[which(as.character(tax_df$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_Source_SMM.txt",
            sep="\t", row.names = F, quote = F)
```
